plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/libraryElement/impl/ConnectionAnnotations.java

```diff
@@ -337,7 +337,7 @@ public static boolean validateVarInOutConnectionsFormsNoLoop(final Connection co
 	}
 
 	private static boolean isDuplicateConnection(final Connection connection) {
-		return connection.getSource().getOutputConnections().stream().anyMatch(
+		return connection.getSource() != null && connection.getSource().getOutputConnections().stream().anyMatch(
```

Comment:
how can a connection source be null, it should be at least to an errormarker pin